### PR TITLE
New section of the book: nightly rust

### PR DESCRIPTION
**Pull request by steveklabnik**
Now that feature flags are only on nightly, it's good to split this stuff out.

### CODE REVIEW

**Comment by steveklabnik** [comment]
This diff is basically just extracting all the parts of the book out, with the exception of `trace_macros`. The rest of it is just verbatim copy/paste.

**Comment by alexcrichton** [comment]
Is the empty file  src/doc/trpl/tracing-macros.md  intentional?

**Comment by steveklabnik** [comment]
Uh, no. Whoops!

File: src/doc/trpl/unsafe.md [hunk 1]
```diff
@@ -181,539 +181,3 @@ code:
...
-Current valid options are:
-
-1. *volatile* - specifying this is analogous to
-   `__asm__ __volatile__ (...)` in gcc/clang.
-2. *alignstack* - certain instructions expect the stack to be
-   aligned a certain way (i.e. SSE) and specifying this indicates to
-   the compiler to insert its usual stack alignment code
-3. *intel* - use intel syntax instead of the default AT&T.
-
-# Avoiding the standard library
```
Comment (alexcrichton) [inline src/doc/trpl/unsafe.md]: This section appears to have disappeared, was that intentional? (I'm ok if it was)

Comment (alexcrichton) [inline src/doc/trpl/unsafe.md]: Oh wait nevermind this header just changed names and my search-fu failed me

**Comment by alexcrichton** [comment]
@bors: r+ rollup

Nice idea!

**Comment by steveklabnik** [comment]
:confetti_ball:

**Comment by erickt** [comment]
I think this errored out in the [rollup](http://buildbot.rust-lang.org/builders/auto-mac-64-opt/builds/4222/steps/compile/logs/stdio) with:

```
rustc: x86_64-apple-darwin/stage2/lib/rustlib/x86_64-apple-darwin/lib/librustc_llvm
rustc: x86_64-apple-darwin/stage1/lib/rustlib/x86_64-apple-darwin/bin/rustbook
rustc: x86_64-apple-darwin/stage1/lib/rustlib/x86_64-apple-darwin/bin/rustdoc
/Users/rustbuild/src/rust-buildbot/slave/auto-mac-64-opt/build/src/driver/driver.rs:15:14: 15:23 warning: obsolete syntax: "crate-name"
/Users/rustbuild/src/rust-buildbot/slave/auto-mac-64-opt/build/src/driver/driver.rs:15 extern crate "rustdoc" as this;
                                                                                                    ^~~~~~~~~
note: use an identifier not in quotes instead
/Users/rustbuild/src/rust-buildbot/slave/auto-mac-64-opt/build/src/driver/driver.rs:18:14: 18:28 warning: obsolete syntax: "crate-name"
/Users/rustbuild/src/rust-buildbot/slave/auto-mac-64-opt/build/src/driver/driver.rs:18 extern crate "rustc_driver" as this;
                                                                                                    ^~~~~~~~~~~~~~
cp: x86_64-apple-darwin/stage2/lib/librustdoc
rustc: x86_64-apple-darwin/stage2/lib/rustlib/x86_64-apple-darwin/lib/libterm
rustc: x86_64-apple-darwin/stage2/lib/rustlib/x86_64-apple-darwin/lib/libserialize
cp: x86_64-apple-darwin/stage2/bin/rustbook
cp: x86_64-apple-darwin/stage2/bin/rustdoc
rustbook: doc/book/index.html
error: No such file or directory (os error 2)
make: *** [doc/book/index.html] Error 101
make: *** Waiting for unfinished jobs....
program finished with exit code 2
elapsedTime=2077.192005
```

**Comment by steveklabnik** [comment]
I don't know what caused this error, but I'll rebase in case there's something weird, and let's just non-rollup this one.

**Pull request merged**